_posts/hannahlwang/2016-04-05-hannah-MOAR-dictionary-exercises.md

```diff
@@ -0,0 +1,15 @@
+---
+layout: post
+author: hannahlwang
+title: "Hannah's MOAR Dictionary Exercises"
+---
+
+<h3>Exercise 1</h3>
+
+<iframe src="https://trinket.io/embed/python3/4b98c7cbf5" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+<h3>Exercises 2-5</h3>
+
+<iframe src="https://trinket.io/embed/python3/2c65bafc89" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+I get dictionaries!!! I really enjoyed how we were able to work on these classes by ourselves in class, with the option to consult. This was the best way for me to figure out the content on my own time, and then also ask questions when I ran into problems but felt like I had a clear enough grasp of the concepts. I enjoyed how exercises 2-5 all kind of built off each other. For most of them, I was able to use a dictionary as a counter. It also gave me a chance to practice searching with regexes, which I found really useful. I also decided to go through and comment basically every line after I finished, just to make sure I fully understood what I had done.
```

Comment:
💡 !  Nice work catching up after your sick day.  You did a great job identifying the key data you'd want for the exercise and then regexing only it.  You made your task a lot easier this way.